app/models/migration/ecf/base_record.rb

```diff
@@ -0,0 +1,7 @@
+module Migration::Ecf
+  class BaseRecord < ApplicationRecord
+    self.abstract_class = true
+
+    connects_to database: { reading: :ecf, writing: :ecf } unless Rails.env.review?
```

Comment:
maybe we can use the api separation flag instead here to avoid connecting in prod for now?

Reply:
So its a bit of a difficult one, as if we omit the `connects_to` it will default to the `primary` NPQ DB for the ECF models. We want to do this in the review apps because there is no ECF DB spun up in review, but I figured for prod we can let the connection happen and its usage will be restricted elsewhere (the migration service will have a safety check that looks at the config to ensure its enabled).

We could override `readonly?` to prevent any accidental writes on environments where its disabled though?

https://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-readonly-3F

Reply:
I've made this change [here](https://github.com/DFE-Digital/npq-registration/pull/1185/commits/a94fb076b9bd348a8226d3c040a841c3f13cdba5)